### PR TITLE
Add runbot_custom_demo_data module

### PR DESCRIPTION
**Pull request by niboo-sam**
Add the runbot_custom_demo_data module that allows to create a build with only repositories demo data

It adds a boolean on the repository form view.
If the user checks it, a job will be run to install all the modules needed without demo data.
After that, the normal 20 job is launched, and it will only load demo data from specified modules.

### CODE REVIEW

File: runbot_custom_demo_data/runbot_build.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+_logger = logging.getLogger(__name__)
+
+
+def grep(filename, string):
+    if os.path.isfile(filename):
+        return open(filename).read().find(string) != -1
+    return False
+
+
+def now():
```
Comment (moylop260) [inline runbot_custom_demo_data/runbot_build.py]: You can use original [now](https://github.com/odoo/odoo-extra/blob/80722c03def52c8c1876d78ef3e805531562e8cb/runbot/runbot.py#L121) method with
`from openerp.addons.runbot.runbot import now`

File: runbot_custom_demo_data/runbot_build.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import time
+import logging
+import openerp
+from openerp.fields import Boolean
+from openerp.models import api, Model
+
+_logger = logging.getLogger(__name__)
+
+
+def grep(filename, string):
```
Comment (moylop260) [inline runbot_custom_demo_data/runbot_build.py]: You can use original [grep](https://github.com/odoo/odoo-extra/blob/80722c03def52c8c1876d78ef3e805531562e8cb/runbot/runbot.py#L69) method with
`from openerp.addons.runbot.runbot import grep`

File: runbot_custom_demo_data/runbot_build.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    _inherit = "runbot.build"
+
+    @api.model
+    def job_15_install_all(self, build, lock_path, log_path):
+        """
+        Install needed modules and dependencies without any demo data
+        """
+        if build.repo_id.custom_demo_data:
+            build._log('install_all', 'Start install all modules')
+            self.pg_createdb("%s-all" % build.dest)
```
Comment (moylop260) [inline runbot_custom_demo_data/runbot_build.py:33]: Why you are duplicating database name of [job_20](https://github.com/odoo/odoo-extra/blob/80722c03def52c8c1876d78ef3e805531562e8cb/runbot/runbot.py#L883)?

Comment (niboo-sam) [inline runbot_custom_demo_data/runbot_build.py:33]: I m not duplicating it.
I m creating it only if custom demo data are needed.
In the job_20, I m not creating it again.

Comment (moylop260) [inline runbot_custom_demo_data/runbot_build.py:33]: If you run two times next sentence: `self.pg_createdb("%s-all" % build.dest)`
You will see next error:
`createdb: database creation failed: ERROR:  database "BUILD.DEST-all" already exists`

You are running this command in next sentences:
1. [job_15_install_all](https://github.com/Niboo/runbot-addons/blob/1727088be21112190df7a721278ff20dccda2771/runbot_custom_demo_data/runbot_build.py#L50)
2. [original job_20_test_all](https://github.com/odoo/odoo-extra/blob/80722c03def52c8c1876d78ef3e805531562e8cb/runbot/runbot.py#L883)
3. [custom job_20_test_all](https://github.com/Niboo/runbot-addons/blob/1727088be21112190df7a721278ff20dccda2771/runbot_custom_demo_data/runbot_build.py#L75)

File: runbot_custom_demo_data/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    "name": "Runbot Custom Demo Data",
+    "version": "0.1",
+    "author": "Niboo,Odoo Community Association (OCA)",
+    "category": "Runbot",
+    "description": """
```
Comment (moylop260) [inline runbot_custom_demo_data/__openerp__.py:10]: Can you create and use [README.rst](https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst) template?

File: runbot_custom_demo_data/runbot_build.py (hunk 3 above)
Comment (niboo-sam) [inline runbot_custom_demo_data/runbot_build.py:33]: As the method is inherited, the original is not triggered.

File: runbot_custom_demo_data/__openerp__.py (hunk 4 above)
Comment (niboo-sam) [inline runbot_custom_demo_data/__openerp__.py:10]: Done!

File: runbot_custom_demo_data/runbot_repo.xml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+        <field name="inherit_id" ref="runbot.view_repo_form"/>
+        <field name="name">View Repo Inherit</field>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='duplicate_id']" position="after">
+                <field name="custom_demo_data"/>
+            </xpath>
+        </field>
+    </record>
+</data>
+</openerp>
```
Comment (osvalr) [inline runbot_custom_demo_data/runbot_repo.xml]: EOF is expected

File: runbot_custom_demo_data/README.rst [hunk 6]
```diff
@@ -0,0 +1,61 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Runbot Custom Demo Data
```
Comment (elicoidal) [inline runbot_custom_demo_data/README.rst:7]: @  Thanks for your contribution to the OCA. 
Let me introduce to you some basic information about contribution in OCA and I will review other points in more comments inside the code..
In general, please follow the principles: 
- Contribution guidelines: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md
- Please follow the OCA README template:
  https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst
- We recommend to include a screenshot as well to help new comers (nice to have)
- Please use simple headers in all py files following OCA templates
  https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3
- Please sort all imports in alphabetical order following: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#imports

File: runbot_custom_demo_data/README.rst [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+=======================
+
+This module was written to extend the functionality of runbot to run build with
```
Comment (elicoidal) [inline runbot_custom_demo_data/README.rst]: This module extends the functionality of runbot to run build with

File: runbot_custom_demo_data/README.rst [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+Usage
+=====
+
+To use this module, you need to:
+
+* Check the custom demo data checkbox in repository
+
+
+Known issues / Roadmap
```
Comment (elicoidal) [inline runbot_custom_demo_data/README.rst]: remove

File: runbot_custom_demo_data/README.rst [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+* (empty)
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/runbot-addons/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/runbot-addons/issues/new?body=module:%20runbot_pylint%0Aversion:%208.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (elicoidal) [inline runbot_custom_demo_data/README.rst]: remove

File: runbot_custom_demo_data/README.rst [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
```
Comment (elicoidal) [inline runbot_custom_demo_data/README.rst]: https

File: runbot_custom_demo_data/__init__.py [hunk 11]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
```
Comment (elicoidal) [inline runbot_custom_demo_data/__init__.py:1]: Please use simplified headers

File: runbot_custom_demo_data/__openerp__.py [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    "name": "Runbot Custom Demo Data",
+    "version": "0.1",
```
Comment (elicoidal) [inline runbot_custom_demo_data/__openerp__.py:7]: 8.0.1.0.0

File: runbot_custom_demo_data/runbot_repo.xml (hunk 5 above)
Comment (niboo-sam) [reply runbot_custom_demo_data/runbot_repo.xml]: Done

File: runbot_custom_demo_data/README.rst (hunk 10 above)
Comment (niboo-sam) [reply runbot_custom_demo_data/README.rst]: Done

File: runbot_custom_demo_data/__init__.py (hunk 11 above)
Comment (niboo-sam) [reply runbot_custom_demo_data/__init__.py:1]: Done

**Comment by moylop260** [comment]
You have commits of 9.0 branch for 8.0 branch.

**Comment by niboo-sam** [comment]
I will create a new PR but on V9 as the code has been runned on v9 and I m not sure about the backward compatibility.

**Comment by hbrunn** [comment]
closing due to lack of activity

**Pull request closed without merging**